docs/architecture/adr-001-coin-source-tracing.md

```diff
@@ -239,117 +271,131 @@ func GetTraceHashFromDenom(rawDenom string) (tmbytes.HexBytes, error) {
 }
 ```
 
-When a fungible token is sent to a sink chain, the trace information needs to be updated with the
+When a fungible token is sent to from a source chain, the trace information needs to be stored with the
 new port and channel identifiers:
 
-The denomination also needs to be updated when token is received on the source chain. This is done during the `SendTransfer`'s `createOutgoingPacket` call and the `OnRecvPacket` call. Below is the updated function that sets the :
-
 ```golang
-// createOutgoingPacket
+// SendTransfer
 // ...
-prefix := types.GetDenomPrefix(destinationPort, destinationChannel)
-source := strings.HasPrefix(amount[0].Denom, prefix)
 
-for _, denomTrace := range denomTraces {
+// NOTE: SendTransfer simply sends the denomination as it exists on its own
+// chain inside the packet data. The receiving chain will perform denom
+// prefixing as necessary.
+
+if types.SenderChainIsSource(sourcePort, sourceChannel, token.Denom) {
+  // create the escrow address for the tokens
+  escrowAddress := types.GetEscrowAddress(sourcePort, sourceChannel)
+
+  // escrow source tokens. It fails if balance insufficient.
+  if err := k.bankKeeper.SendCoins(
+    ctx, sender, escrowAddress, sdk.NewCoins(token),
+  ); err != nil {
+    return err
+  }
+} else {
   // set the value to the lookup table if not stored already
```

Comment:
Why would we need to set a trace during `SendPacket`? We only need to do so on RECV correct?